### PR TITLE
** Fix #3333 subjects order in vops

### LINKED ISSUE #3333: Rearrange some of the subjects of VOPS
Aineet ovat edelleen väärässä järjestyksessä VOPS:sissa. Tässä ainejärjestys:
ÄI / S2
RUB
ENA + muut vieraat kielet
MAY
MAA / MAB
BI
GE
FY
KE
FI
PS
HI
YH
UE / UO / ET
TE
LI
MU
KU
KÄ
ME
AT
OP
TO
MUU (esim. Tulevaisuuden tutkimus)

### PR DESCRIPTION
**Pull request by ilmoeuro**
** Fixes #3333

### CODE REVIEW

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java [hunk 1]
```diff
@@ -41,7 +45,39 @@
...
   
+  @Inject
+  private Logger log;
+  
   private static final Pattern UPPER_SECONDARY_SCHOOL_SUBJECT_PATTERN = Pattern.compile("^[A-ZÅÄÖ0-9]+$");
+  
+  @PostConstruct
+  public void init() {
+    String commaSeparatedSubjectsOrder = pluginSettingsController.getPluginSetting("transcriptofrecords", "subjectsOrder");
+    if (commaSeparatedSubjectsOrder == null) {
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java]: StringUtils.isBlank/isEmpty? Or is it by design that there's no warning if the pluginSetting is empty?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java [hunk 2]
```diff
@@ -41,7 +45,39 @@
...
+      log.log(Level.WARNING, "No subjects order defined for studies vies");
+      return;
+    }
+
+    String[] subjects = commaSeparatedSubjectsOrder.split(",");
+    int i = 0;
+    for (String subject : Arrays.asList(subjects)) {
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java]: Maybe just a plain old for(int i..? Would be 2 rows less

Comment (ilmoeuro) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java]: I don't think saving a few lines is worth losing the safety benefits of iterator-based for. When indexing an array, there's always a possibility of indexing out of bounds, so I try to minimize indexing.

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java (hunk 1 above)
Comment (ilmoeuro) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java]: Good point, I'll add an isBlank check.

**Pull request merged**